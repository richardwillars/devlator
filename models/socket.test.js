const socketModule = require('./socket');

describe('models/socket', () => {
  it('should export the schema', () => {
    const constants = {
      socket: {
        a: 'A',
        b: 'B',
        c: 'C',
        d: 'D',
      },
    };
    const events = {
      a: 'A',
      b: 'B',
      c: 'C',
      d: 'D',
      e: 'E',
      f: 'F',
      g: 'G',
    };

    const mongoose = {
      model: () => {},
      Schema: class Schema {},
    };
    const socketModel = socketModule(mongoose, events, constants);
    expect(socketModel.schema).toMatchSnapshot();
  });

  it('should export the model', () => {
    const constants = {
      socket: {
        a: 'A',
        b: 'B',
        c: 'C',
        d: 'D',
      },
    };
    const events = {
      a: 'A',
      b: 'B',
      c: 'C',
      d: 'D',
      e: 'E',
      f: 'F',
      g: 'G',
    };

    const modelMock = {};
    const mongoose = {
      model: jest.fn().mockReturnValue(modelMock),
      Schema: class Schema {
        constructor(props) {
          this.props = props;
        }
      },
    };
    const socketModel = socketModule(mongoose, events, constants);

    expect(socketModel.model).toEqual(modelMock);
    expect(mongoose.model).toHaveBeenCalledTimes(1);
    expect(mongoose.model.mock.calls[0][0]).toEqual('Socket');
    const initialisedSchema = mongoose.model.mock.calls[0][1];
    expect(initialisedSchema instanceof mongoose.Schema).toEqual(true);
    expect(initialisedSchema.props).toMatchSnapshot();
  });
});
