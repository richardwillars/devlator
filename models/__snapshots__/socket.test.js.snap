// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/socket should export the model 1`] = `
Object {
  "_id": false,
  "additionalInfo": Object {
    "default": Object {},
    "required": false,
    "type": [Function],
  },
  "commands": Object {
    "off": Object {
      "description": "Turns the socket off",
      "friendly": "Turn off",
      "type": [Function],
    },
    "on": Object {
      "description": "Turns the socket on",
      "friendly": "Turn on",
      "type": [Function],
    },
  },
  "events": Object {
    "a": "A",
    "b": "B",
    "c": "C",
    "d": "D",
  },
  "name": Object {
    "required": true,
    "type": [Function],
  },
  "originalId": Object {
    "required": true,
    "type": [Function],
  },
}
`;

exports[`models/socket should export the schema 1`] = `
Object {
  "_id": false,
  "additionalInfo": Object {
    "default": Object {},
    "required": false,
    "type": [Function],
  },
  "commands": Object {
    "off": Object {
      "description": "Turns the socket off",
      "friendly": "Turn off",
      "type": [Function],
    },
    "on": Object {
      "description": "Turns the socket on",
      "friendly": "Turn on",
      "type": [Function],
    },
  },
  "events": Object {
    "a": "A",
    "b": "B",
    "c": "C",
    "d": "D",
  },
  "name": Object {
    "required": true,
    "type": [Function],
  },
  "originalId": Object {
    "required": true,
    "type": [Function],
  },
}
`;
